plugins/serializers/customjson/customjson.go

```diff
@@ -0,0 +1,168 @@
+package customjson
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf"
+	"github.com/jmespath/go-jmespath"
+)
+
+type serializer struct {
+	JmespathExpression string
+	TagsPrefix         string
+}
+
+func NewSerializer(jmespath_expression string, tags_prefix string) (*serializer, error) {
+	s := &serializer{
+		JmespathExpression: jmespath_expression,
+		TagsPrefix:         tags_prefix,
+	}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+
+	m, err := s.createObject(metric)
+	if err != nil {
+		return nil, fmt.Errorf("D! [serializer.customjson] Dropping invalid metric: %s", metric.Name())
+	}
+
+	return m, nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+
+	var serialized []byte
+
+	for _, metric := range metrics {
+		m, err := s.createObject(metric)
+		if err != nil {
+			return nil, fmt.Errorf("D! [serializer.customjson] Dropping invalid metric: %s", metric.Name())
```

Comment:
Same as above. Please rethink if you want to stop serialization if one metric is invalid or not. Both is fine for me...